src/main/java/org/zalando/nakadi/repository/kafka/KafkaLocationManager.java

```diff
@@ -24,73 +27,34 @@
     private final ZooKeeperHolder zkFactory;
     private final Properties kafkaProperties;
     private final KafkaSettings kafkaSettings;
+    private final ScheduledExecutorService scheduledExecutor;
 
     public KafkaLocationManager(final ZooKeeperHolder zkFactory, final KafkaSettings kafkaSettings) {
         this.zkFactory = zkFactory;
-        this.kafkaProperties = buildKafkaProperties(fetchBrokers());
+        this.kafkaProperties = new Properties();
         this.kafkaSettings = kafkaSettings;
+        this.scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
+        this.scheduledExecutor.scheduleAtFixedRate(this::updateBootstrapServers, 0, 1, TimeUnit.MINUTES);
```

Comment:
For the record - we are relying on a fact that KafkaClient will throw exception in case if bootstrap servers are empty.